### PR TITLE
PQKmeans can be pickled!

### PR DESCRIPTION
**Pull request by kogaki**
Implemented `__getstate__` and `__setstate__` on PQKmeans following the way #17 .
So now we can pickle PQKmeans.

### CODE REVIEW

File: test/clustering/test_pqkmeans.py [hunk 1]
```diff
@@ -60,11 +62,10 @@ def test_constructor_with_cluster_center(self):
         cluster_centers = numpy.array(engine.cluster_centers_, dtype=numpy.uint8)
         predicted = engine.predict(codes)
 
-        # Create new pqkmeans, but construct with initial centers.
-        engine_recovered = pqkmeans.clustering.PQKMeans(encoder=self.encoder, k=5, iteration=10, verbose=False, init_centers=cluster_centers)
+
+        # save current engine and recover from savedata
+        engine_savedata = pickle.dumps(engine)
+        engine_recovered = pickle.loads(engine_savedata)
```
Comment (kogaki) [inline test/clustering/test_pqkmeans.py]: Just pickle the trained engine instead of creating new class

File: pqkmeans/clustering/pqkmeans.py [hunk 2]
```diff
@@ -1,37 +1,52 @@
...
 
 class PQKMeans(sklearn.base.BaseEstimator, sklearn.base.ClusterMixin):
-    def __init__(self, encoder, k, iteration=10, verbose=False, init_centers=None):
+    def __init__(self, encoder, k, iteration=10, verbose=False):
         super(PQKMeans, self).__init__()
-        if init_centers is None:
-            init_centers = []
-        self._impl = _pqkmeans.PQKMeans(encoder.codewords, k, iteration, verbose, init_centers)
+        self.encoder = encoder
+        self._impl = _pqkmeans.PQKMeans(self.encoder.codewords, k, iteration, verbose)
```
Comment (kogaki) [inline pqkmeans/clustering/pqkmeans.py]: Sorry, I removed `init_centers` to unify the way of loading trained instance. @insikk

File: pqkmeans/clustering/pqkmeans.py [hunk 3]
```diff
@@ -1,37 +1,52 @@
...
     def predict(self, x_test):
         # type: (numpy.array) -> Any
         assert len(x_test.shape) == 2
         return numpy.array(list(self.predict_generator(x_test)))
 
+    def __getstate__(self):
+        return PQKMeansSavedata(
+            encoder=self.encoder,
+            k=self._impl.k_,
+            iteration=self._impl.k_,
```
Comment (matsui528) [inline pqkmeans/clustering/pqkmeans.py]: This should be `self._impl.iteration_`

Comment (kogaki) [reply pqkmeans/clustering/pqkmeans.py]: Oh sorry, i've fixed it.

**Comment by matsui528** [comment]
Now we finally solved #12 and #16 . I'll update a README and a tutorial then. Thanks again for  @insikk !

**Pull request merged**